arches/app/search/es_mapping_modifier.py

```diff
@@ -0,0 +1,80 @@
+from arches.app.models.system_settings import settings
+from arches.app.utils import import_class_from_string
+
+
+class EsMappingModifierFactory:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def get_es_mapping_modifier_classes():
+        return (
+            [
+                import_class_from_string(classname)
+                for classname in settings.ES_MAPPING_MODIFIER_CLASSES
+            ]
+            if settings.setting_exists("ES_MAPPING_MODIFIER_CLASSES")
+            and settings.ES_MAPPING_MODIFIER_CLASSES
+            else []
+        )
+
+
+class EsMappingModifier:
+    """
+    Base class for creating custom sections in the Resource Instance elasticsearch document.
+    """
+
+    custom_search_path = "custom_values"
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def get_custom_search_path():
+        """
+        Identifies the document key where the custom ES document section is located.
+
+        :return: ES document key
+        :rtype String
+        """
+        return EsMappingModifier.custom_search_path
+
+    @staticmethod
+    def add_search_terms(resourceinstance, document, terms):
+        """
+        Adds the custom ES search document section for the resource instance.
+        :param resourceinstance: resource instance being indexed
+        :param document: Original ES document for the Resource Instance
+        :param terms: ES terms in the document
+        """
+        pass
+
+    @staticmethod
+    def add_search_filter(search_query, term):
+        """
+        Adds to or modifies the term search_query to include the custom search document section as part of the search
+        :param search_query: The original search term query
+        :param term: The search term
+        """
+        pass
+
+    @staticmethod
+    def get_custom_search_config():
```

Comment:
```suggestion
    def get_mapping_definition():
```

see this [doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping.html) on the mapping definition